### PR TITLE
Add Redux Toolkit to bookstore app

### PR DESCRIPTION
**Pull request by martinkarugaba**
# :factory_worker: What I have worked on
- [ ] Configure a Redux store.
- [ ] Added two slices books and categories slices.
- [ ] Defined a slice of state for books and categories.
- [ ] Added the required reducers.

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @martinkarugaba , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that you include all the headers from the [MIcroverse template](https://github.com/microverseinc/readme-template). Currently I saw that you are missing the future features  header.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/category/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+  underConstruction: false,
+};
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
```
Comment (zainsadaqat) [inline src/redux/category/categoriesSlice.js:10]: ### Kindly implement a reducer for the categories that checks the status as it's part of project requirement 👍 
Screenshot 2023-06-14 at 2 34 11 PM

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
## Hello @martinkarugaba  👋 🙂

You're doing a great job on your project. We're proud of you. Good job so far! 

It's OK to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job. 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Flow
  ✅  Descriptive Pull Request
  ✅  No linter errors

## Required Changes ♻️
Due to some issues in GitHub, the asked changes in the bookSlice file aren't being reflected here that's why I've to mention it here.
### Kindly implement the reducers for adding and removing a book as it's part of the project requirement. 👍 
Screenshot 2023-06-14 at 2 44 17 PM




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Martin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**